src/Sparrow/Utils/TypeCache.cs

```diff
@@ -92,4 +102,80 @@ private int GetBucket(Type type)
             return hashCode % _size;
         }
     }
+
+    /// <summary>
+    /// A lightweight, hash-based cache for associating .NET <see cref="Type"/> objects with values of a generic type <typeparamref name="T"/>.
+    /// This cache prioritizes performance over strict synchronization, making it suitable for scenarios where occasional cache misses
+    /// are acceptable but incorrect (Type, T) mappings must never occur.
+    /// </summary>
+    internal sealed class ReplacementTypeCache<T>
```

Comment:
Where is this being used?

Reply:
This is used in the Object Parser.
https://github.com/ravendb/ravendb/pull/20025/commits/b7885aa94c988c42181773d8372bd869b410e5a9#diff-faa77283078aa5534a42bf3d5c2c49634bbd5b635cf27dc677826bf49eb2d809R575

Reply:
Uses `TypeCache`, no `ReplacementTypeCache`

Reply:
No. https://github.com/ravendb/ravendb/pull/20025/commits/02d06c8722dbb5bd1774bcb89b2ef65f0af4ffd9#diff-faa77283078aa5534a42bf3d5c2c49634bbd5b635cf27dc677826bf49eb2d809R328
Am I missing something?

Reply:
Previous link didn't have it, nvm - all okay now